Unable to compile when adding Vulkan Loader's CMake as a dependency on a project with no git repository

**Describe the bug**
You get a compiler error when following the steps below.
I'll describe why I think the issue happens after listing the repro steps, otherwise it doesn't make a lot of sense. I also propose a couple of fixes.

**Environment:**
 - OS: Windows 10 (10.0.19043.1645), 64 bit
 - GPU: Nvidia Geforce GTX950M
 - Graphics Driver: Nvidia 512.15
 - Header version: [v1.3.212](https://github.com/KhronosGroup/Vulkan-Headers/releases/tag/v1.3.212)
 - Compiler: MSVC 19.29.30140, CMake 3.23.1

**To Reproduce**
Steps to reproduce the behavior:
1. Create a dummy project with a CMakeLists.txt but don't initialize git for that project folder.
2. Use `git clone` on the [Vulkan-Headers](https://github.com/KhronosGroup/Vulkan-Headers.git) and [Vulkan-Loader](https://github.com/KhronosGroup/Vulkan-Loader.git) repositories (into a folder called `external/` in my example).
3. Add both as dependencies for your project in CMakeLists.txt (an example one will be provided below)
4. If you now run cmake (`cmake -S . -B .build/`) you'll see this: `fatal: not a git repository (or any of the parent directories): .git`. 
5. You can ignore this for now, but if you try to compile from the root of your project (`cmake --build .build/`) you'll get this error:
```
<MyPath>\external\Vulkan-Loader\loader\loader.c(1550,33): error C2220: the following warning is treated as an error [<MyPath>\.build\external\Vulkan-Loader\loader\vulkan.vcxproj]
<MyPath>\external\Vulkan-Loader\loader\loader.c(1550,33): warning C4003: not enough arguments for function-like macro invocation 'LOADER_GIT_STRINGIFY' [<MyPath>\.build\external\Vulkan-Loader\loader\vulkan.vcxproj]
<MyPath>\external\Vulkan-Loader\loader\loader.c(1550,66): error C2059: syntax error: ';' [<MyPath>\.build\external\Vulkan-Loader\loader\vulkan.vcxproj]
```

Example CMakeLists.txt (this should be enough):
```cmake
cmake_minimum_required(VERSION 3.21)
project(ExampleProject)

set(EXTERNAL_PATH ${PROJECT_SOURCE_DIR}/external)

# Vulkan-Headers
add_subdirectory(${EXTERNAL_PATH}/Vulkan-Headers)

# Vulkan-Loader
add_subdirectory(${EXTERNAL_PATH}/Vulkan-Loader)

# Code
# Get your source files here into SOURCE_FILES

# Project
add_executable(${PROJECT_NAME} ${SOURCE_FILES})
set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 17)
target_link_libraries(${PROJECT_NAME} vulkan)
```


**Reason for the bug**
I might be wrong but this is what I think is happening.
I believe the reason for the bug comes from Vulkan-Loader's CMakeLists.txt.
You'll see that this if statement is true: `if (GIT_FOUND AND EXISTS "${PROJECT_SOURCE_DIR}/.git/HEAD")`.
It's true because `${PROJECT_SOURCE_DIR}` will be `<MyPath>/external/Vulkan-Loader/` and in that directory there will be a `.git` folder as per repro step nr 2.
However, the next line will fail. 
```cmake
execute_process(
    COMMAND ${GIT_EXECUTABLE} describe --tags --always
    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
    OUTPUT_VARIABLE GIT_TAG_INFO)
```
The line fails in repro step nr 4, because here you're using `CMAKE_SOURCE_DIR` as the working directory which will be `<MyPath>/`. Remember, that repro step nr 1 says not to initialize a git repository, so in that directory there's no `.git` folder. Because of that, `GIT_TAG_INFO` will be empty and cause the compiler error in repro step nr 5.


**Suggested fix**
Before I suggest the fix, let me say that I don't fully understand why this exists in the first place. It seems to be used for a custom log in `loader.c`: 
```c
loader_log(NULL, VULKAN_LOADER_INFO_BIT, 0, "[Git - Tag: %s, Branch/Commit: %s]", git_tag_info, git_branch_name);
```
I don't understand why we check for `PROJECT_SOURCE_DIR` while using `CMAKE_SOURCE_DIR` in the command.
I believe your intention was to use `PROJECT_SOURCE_DIR` for the working directory in that command.
Or, my preferred solution, we could add an option to disable these shenanigans altogether :)
Or both!

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -148,15 +148,17 @@ if(APPLE)
     set(CMAKE_MACOSX_RPATH 1)
 endif()
 
+set(GIT_BRANCH_NAME "--unknown--")
+set(GIT_TAG_INFO "--unknown--")
 find_package (Git)
-if (GIT_FOUND AND EXISTS "${PROJECT_SOURCE_DIR}/.git/HEAD")
+if (GIT_FOUND AND EXISTS "${CMAKE_CURRENT_LIST_DIR}/.git/HEAD")
     execute_process(
         COMMAND ${GIT_EXECUTABLE} describe --tags --always
-        WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+        WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}
         OUTPUT_VARIABLE GIT_TAG_INFO)
     string(REGEX REPLACE "\n$" "" GIT_TAG_INFO "${GIT_TAG_INFO}")
 
-    file(READ ".git/HEAD" GIT_HEAD_REF_INFO)
+    file(READ "${CMAKE_CURRENT_LIST_DIR}/.git/HEAD" GIT_HEAD_REF_INFO)
     if (GIT_HEAD_REF_INFO)
         string(REGEX MATCH "ref: refs/heads/(.*)" _ ${GIT_HEAD_REF_INFO})
         if (CMAKE_MATCH_1)
@@ -165,12 +167,7 @@ if (GIT_FOUND AND EXISTS "${PROJECT_SOURCE_DIR}/.git/HEAD")
             set(GIT_BRANCH_NAME ${GIT_HEAD_REF_INFO})
         endif()
         string(REGEX REPLACE "\n$" "" GIT_BRANCH_NAME "${GIT_BRANCH_NAME}")
-    else()
-        set(GIT_BRANCH_NAME "--unknown--")
     endif()
-else()
-    set(GIT_BRANCH_NAME "--unknown--")
-    set(GIT_TAG_INFO "--unknown--")
 endif()
 
 if(WIN32 AND CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
@@ -341,7 +338,7 @@ endif()
 
 # DEBUG enables runtime loader ICD verification
 # Add git branch and tag info in debug mode
-target_compile_definitions(loader_common_options INTERFACE $<$<CONFIG:DEBUG>:DEBUG;GIT_BRANCH_NAME=${GIT_BRANCH_NAME};GIT_TAG_INFO=${GIT_TAG_INFO}>)
+target_compile_definitions(loader_common_options INTERFACE $<$<CONFIG:DEBUG>:DEBUG;GIT_BRANCH_NAME="${GIT_BRANCH_NAME}";GIT_TAG_INFO="${GIT_TAG_INFO}">)
 
 # Check for the existance of the secure_getenv or __secure_getenv commands
 include(CheckFunctionExists)

--- a/loader/loader.c
+++ b/loader/loader.c
@@ -1544,11 +1544,7 @@ void loader_initialize(void) {
     loader_log(NULL, VULKAN_LOADER_INFO_BIT, 0, "Vulkan Loader Version %d.%d.%d", version.major, version.minor, version.patch);
 
 #if defined(GIT_BRANCH_NAME) && defined(GIT_TAG_INFO)
-#define LOADER_GIT_STRINGIFY(x) #x
-#define LOADER_GIT_TOSTRING(x) LOADER_GIT_STRINGIFY(x)
-    const char git_branch_name[] = LOADER_GIT_TOSTRING(GIT_BRANCH_NAME);
-    const char git_tag_info[] = LOADER_GIT_TOSTRING(GIT_TAG_INFO);
-    loader_log(NULL, VULKAN_LOADER_INFO_BIT, 0, "[Git - Tag: %s, Branch/Commit: %s]", git_tag_info, git_branch_name);
+    loader_log(NULL, VULKAN_LOADER_INFO_BIT, 0, "[Vulkan Loader Git - Tag: " GIT_BRANCH_NAME ", Branch/Commit: " GIT_TAG_INFO "]");
 #endif
 }
 

